aws/resource_aws_network_acl_association.go

```diff
@@ -0,0 +1,140 @@
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsNetworkAclAssociation() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsNetworkAclAssociationCreate,
+		Read:   resourceAwsNetworkAclAssociationRead,
+		Update: resourceAwsNetworkAclAssociationUpdate,
+		Delete: resourceAwsNetworkAclAssociationDelete,
+
+		Schema: map[string]*schema.Schema{
+			"subnet_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"network_acl_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+		},
+	}
+}
+
+func resourceAwsNetworkAclAssociationCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ec2conn
+
+	naclId := d.Get("network_acl_id").(string)
+	subnetId := d.Get("subnet_id").(string)
+
+	log.Printf(
+		"[INFO] Creating network acl association: %s => %s",
+		subnetId,
+		naclId)
+
+	association, err_association := findNetworkAclAssociation(subnetId, conn)
+	if err_association != nil {
+		return fmt.Errorf("Failed to create acl %s with nacl %s: %s", d.Id(), naclId, err_association)
+	}
+
+	associationOpts := ec2.ReplaceNetworkAclAssociationInput{
+		AssociationId: association.NetworkAclAssociationId,
+		NetworkAclId:  aws.String(naclId),
+	}
+
+	var err error
+	err = resource.Retry(5*time.Minute, func() *resource.RetryError {
+		_, err = conn.ReplaceNetworkAclAssociation(&associationOpts)
+		if err != nil {
+			if awsErr, ok := err.(awserr.Error); ok {
+				if awsErr != nil {
+					return resource.RetryableError(awsErr)
+				}
+			}
+			return resource.NonRetryableError(err)
+		}
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+
+	// Set the ID and return
+	d.SetId(naclId)
+	log.Printf("[INFO] Association ID: %s", d.Id())
+
+	return resourceAwsNetworkAclAssociationRead(d, meta)
+}
+
+func resourceAwsNetworkAclAssociationRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ec2conn
+
+	// Inspect that the association exists
+	subnetId := d.Get("subnet_id").(string)
+	_, err_association := findNetworkAclAssociation(subnetId, conn)
+	if err_association != nil {
+		log.Printf("[INFO] Association for %s was not found.", subnetId)
+		d.SetId("")
+		return nil
+	}
+
+	return nil
+}
+
+func resourceAwsNetworkAclAssociationUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ec2conn
+
+	naclId := d.Get("network_acl_id").(string)
+	subnetId := d.Get("subnet_id").(string)
+
+	log.Printf(
+		"[INFO] Creating network acl association: %s => %s",
+		subnetId,
+		naclId)
+
+	association, err_association := findNetworkAclAssociation(subnetId, conn)
+	if err_association != nil {
+		return fmt.Errorf("Failed to update acl %s with subnet %s: %s", d.Id(), naclId, err_association)
+	}
+
+	req := &ec2.ReplaceNetworkAclAssociationInput{
+		AssociationId: association.NetworkAclAssociationId,
+		NetworkAclId:  aws.String(naclId),
+	}
+	resp, err := conn.ReplaceNetworkAclAssociation(req)
```

Comment:
Could you move the log instruction from line 42 here, put it all on one line, and perhaps change it to the below?:

```
log.Printf("[DEBUG] Updating Network ACL association: %#v", associationOpts)
```

This would provide all of the parameters for the function, so that we don't need to Printf them all manually :)